diff --git a/node_modules/react-native-audio-recorder-player/android/build.gradle b/node_modules/react-native-audio-recorder-player/android/build.gradle
index 38cbee2..bd5e678 100644
--- a/node_modules/react-native-audio-recorder-player/android/build.gradle
+++ b/node_modules/react-native-audio-recorder-player/android/build.gradle
@@ -15,7 +15,6 @@ buildscript {
     }
 
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.2"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${getExtOrDefault('kotlinVersion')}"
     }
 }
@@ -28,20 +27,21 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-  compileSdkVersion safeExtGet('compileSdkVersion', 31)
+  namespace "com.dooboolab.audiorecorderplayer"
+  compileSdkVersion safeExtGet('compileSdkVersion', 34)
 
   defaultConfig {
     minSdkVersion safeExtGet('minSdkVersion', 24)
-    targetSdkVersion safeExtGet('targetSdkVersion', 31)
+    targetSdkVersion safeExtGet('targetSdkVersion', 34)
     versionCode 1
     versionName "1.0"
   }
 
   kotlinOptions {
-      jvmTarget = "1.8"
+      jvmTarget = "17"
   }
 
-  lintOptions {
+  lint {
     abortOnError false
   }
 }
@@ -52,21 +52,11 @@ repositories {
 
 dependencies {
   implementation 'com.facebook.react:react-native:+'
-  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${safeExtGet('kotlinVersion', project.ext.kotlinVersion)}"
+  implementation "org.jetbrains.kotlin:kotlin-stdlib:${safeExtGet('kotlinVersion', project.ext.kotlinVersion)}"
 
-  def supportLibVersion = safeExtGet('supportLibVersion', safeExtGet('supportVersion', null))
-  def androidXVersion = safeExtGet('androidXVersion', null)
-  if (supportLibVersion && androidXVersion == null) {
-    implementation "com.android.support:support-annotations:$supportLibVersion"
-    implementation "com.android.support:customtabs:$supportLibVersion"
-  } else {
-    def defaultAndroidXVersion = "1.2.0"
-    if (androidXVersion == null) {
-      androidXVersion = defaultAndroidXVersion
-    }
-    def androidXAnnotation = safeExtGet('androidXAnnotation', androidXVersion)
-    def androidXBrowser = safeExtGet('androidXBrowser', androidXVersion)
-    implementation "androidx.annotation:annotation:$androidXAnnotation"
-    implementation "androidx.browser:browser:$androidXBrowser"
-  }
+  def androidXVersion = safeExtGet('androidXVersion', "1.2.0")
+  def androidXAnnotation = safeExtGet('androidXAnnotation', androidXVersion)
+  def androidXBrowser = safeExtGet('androidXBrowser', androidXVersion)
+  implementation "androidx.annotation:annotation:$androidXAnnotation"
+  implementation "androidx.browser:browser:$androidXBrowser"
 }
diff --git a/node_modules/react-native-audio-recorder-player/android/src/main/java/com/dooboolab.audiorecorderplayer/RNAudioRecorderPlayerModule.kt b/node_modules/react-native-audio-recorder-player/android/src/main/java/com/dooboolab.audiorecorderplayer/RNAudioRecorderPlayerModule.kt
index 02eb743..c1e02eb 100644
--- a/node_modules/react-native-audio-recorder-player/android/src/main/java/com/dooboolab.audiorecorderplayer/RNAudioRecorderPlayerModule.kt
+++ b/node_modules/react-native-audio-recorder-player/android/src/main/java/com/dooboolab.audiorecorderplayer/RNAudioRecorderPlayerModule.kt
@@ -44,13 +44,13 @@ class RNAudioRecorderPlayerModule(private val reactContext: ReactApplicationCont
                 if (Build.VERSION.SDK_INT < 29 &&
                         (ActivityCompat.checkSelfPermission(reactContext, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED ||
                         ActivityCompat.checkSelfPermission(reactContext, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED))  {
-                    ActivityCompat.requestPermissions((currentActivity)!!, arrayOf(
+                    ActivityCompat.requestPermissions((getCurrentActivity())!!, arrayOf(
                             Manifest.permission.RECORD_AUDIO,
                             Manifest.permission.WRITE_EXTERNAL_STORAGE), 0)
                     promise.reject("No permission granted.", "Try again after adding permission.")
                     return
                 } else if (ActivityCompat.checkSelfPermission(reactContext, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
-                    ActivityCompat.requestPermissions((currentActivity)!!, arrayOf(Manifest.permission.RECORD_AUDIO), 0)
+                    ActivityCompat.requestPermissions((getCurrentActivity())!!, arrayOf(Manifest.permission.RECORD_AUDIO), 0)
                     promise.reject("No permission granted.", "Try again after adding permission.")
                     return
                 }
@@ -217,7 +217,7 @@ class RNAudioRecorderPlayerModule(private val reactContext: ReactApplicationCont
                         val key = iterator.nextKey()
                         headers.put(key, httpHeaders.getString(key))
                     }
-                    mediaPlayer!!.setDataSource(currentActivity!!.applicationContext, Uri.parse(path), headers)
+                    mediaPlayer!!.setDataSource(reactContext.applicationContext, Uri.parse(path), headers)
                 } else {
                     mediaPlayer!!.setDataSource(path)
                 }
